### PR TITLE
Add remove branch

### PR DESCRIPTION
**Pull request by MohamedHNoor**
## Created  a new functionality file for:
- Adding a new task
- Deleting a task
- Editing task descriptions

 > All changes to the app lists are saved in local storage

### CODE REVIEW

File: src/functionality.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  input.value = todoArr[todoId].description;
+  editTodoId = todoId;
+};
+
+const deleteTodo = (todoId) => {
+  todoArr = todoArr.filter((todo, index) => index !== todoId);
+  editTodoId = -1;
+  renderTodo();
+  localStorage.setItem('TODO', JSON.stringify(todoArr));
+};
```
Comment (mahbubul14) [inline src/functionality.js]: - [ ] Currently, the indexing number is not updating after deleting a task. Kindly work on it. :+1: 
`As per the project requiement:`

> Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1).

indexing error

File: src/functionality.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  todolistContainer.innerHTML = '';
+
+  // Render todos
+  todoArr.forEach((todo, index) => {
+    todolistContainer.innerHTML += `
+  <div class="list" id=${index}>
+    <input type="checkbox" class="checkbox" job="complete">
+    <p class="text" data-action="edit">${todo.description}</p>
+    <i class="fa-regular fa-trash-can trash" data-action="delete" ></i>
+</div>
```
Comment (mahbubul14) [inline src/functionality.js:46]: - [ ] As per project requirement, the property index is set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal 5). Kindly work on it. :+1: 

indexing error_1

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @MohamedHNoor,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by omarsalem7 — APPROVED** [review]
Hi @MohamedHNoor 😊, 

# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper description ✔️
- The project has met all the technical aspects you needed for this project. Keep rocking!  ✔️


Cheers and Happy coding!👏👏👏

## Optional suggestions

- [ ] [OPTIONAL] It will be better [these guidelines](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) when writing a commit message 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**